storage/innobase/handler/handler0alter.cc

```diff
@@ -10161,6 +10285,13 @@ commit_try_norebuild(
 		}
 	}
 
+	if ((ha_alter_info->handler_flags & ALTER_COLUMN_EQUAL_PACK_LENGTH)
+	    && change_charset_instant_try(
+		trx, ha_alter_info->alter_info->create_list, ctx->new_table,
+		*ha_alter_info->create_info)) {
+		DBUG_RETURN(true);
+	}
+
 	DBUG_RETURN(innobase_instant_try(ha_alter_info, ctx, altered_table,
 					 old_table, trx));
```

Comment:
[Like I wrote on November 9](https://github.com/MariaDB/server/pull/921/files#r232204607), I’d rather construct `ctx->instant_table` with all the changes and then let `innobase_instant_try()` update `SYS_COLUMNS` and other tables. We should also follow this approach for renaming and extending columns, and remove existing functions like `innobase_rename_or_enlarge_columns_cache()`.